aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -249,6 +254,81 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    fun verify_key_rotation_signature(scheme: vector<u8>, public_key_bytes: vector<u8>, signature: vector<u8>, challenge: RotationProofChallenge) : vector<u8> {
+        let auth_key: vector<u8>;
+        if (scheme == ED25519_SCHEME) {
+            let pk = ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = ed25519::new_signature_from_bytes(signature);
+            assert!(ed25519::signature_verify_strict_t(&sig, &pk, challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        } else {
+            let pk = multi_ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = multi_ed25519::new_signature_from_bytes(signature);
+            assert!(multi_ed25519::signature_verify_strict_t(&sig, &pk, challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        };
+        auth_key
+    }
+
+    fun get_challenge_message(account_address: address, in_scheme: vector<u8>, from_public_key_bytes: vector<u8>, to_public_key_bytes: vector<u8>): RotationProofChallenge acquires Account {
```

Comment:
The name `get_challenge_message` is not sufficiently informative here because this function also checks that the input "from PK" is indeed the current PK set at the input account address.

This needs to be renamed, or this functionality should be split out into a different function.

Reply:
I would opt for separating the functionality.